src/test/java/com/suchocki/parkingmeter/ParkingRestControllerTest.java

```diff
@@ -0,0 +1,403 @@
+	private static Driver disabledDriver;
+	private static Driver regularDriverAlreadyParking;
+	private static Driver disabledDriverAlreadyParking;
+
+	private static String regularDriverJSON;
+	private static String disabledDriverJSON;
+	private static String regularDriverAlreadyParkingJSON;
+	private static String disabledDriverAlreadyParkingJSON;
+
+	private static ParkAction parkingRegularDriverParkAction;
+	private static ParkAction parkingDisabledDriverParkAction;
+
+	private static ObjectMapper mapper;
+
+	@BeforeClass
+	public static void prepareObjects() {
+		plnCurrency = new Currency("zloty", "PLN");
+
+		regularPLNCost = new ParkingCost(plnCurrency, 1D, 2D, 1.5);
+		disabledPLNCost = new ParkingCost(plnCurrency, 0D, 2D, 1.2);
+
+		regularType = new DriverType("regular");
+		regularType.addCost(regularPLNCost);
+
+		disabledType = new DriverType("disabled");
+		disabledType.addCost(disabledPLNCost);
+
+		regularDriver = new Driver("wf22222", regularType);
+		disabledDriver = new Driver("wf11111", disabledType);
+		regularDriverAlreadyParking = new Driver("wf33333", regularType);
+		disabledDriverAlreadyParking = new Driver("wf44444", disabledType);
+
+		mapper = new ObjectMapper();
+
+		try {
+			regularDriverJSON = mapper.writeValueAsString(regularDriver);
+			disabledDriverJSON = mapper.writeValueAsString(disabledDriver);
+			regularDriverAlreadyParkingJSON = mapper.writeValueAsString(regularDriverAlreadyParking);
+			disabledDriverAlreadyParkingJSON = mapper.writeValueAsString(disabledDriverAlreadyParking);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Before
+	public void prepareFakeDatabase() {
+		FakeDatabaseStub.currencies.clear();
+		FakeDatabaseStub.drivers.clear();
+		FakeDatabaseStub.parkActions.clear();
+		FakeDatabaseStub.driverPayments.clear();
+
+		FakeDatabaseStub.currencies.add(plnCurrency);
+
+		FakeDatabaseStub.drivers.add(regularDriver);
+		FakeDatabaseStub.drivers.add(disabledDriver);
+		FakeDatabaseStub.drivers.add(regularDriverAlreadyParking);
+		FakeDatabaseStub.drivers.add(disabledDriverAlreadyParking);
+
+	}
+
+	/*------------starting parking meter------------*/
+	@Test
+	public void shouldStartParkingMeterRegularDriver() {
+
+		try {
+			mvc.perform(post("/api/start").contentType(MediaType.APPLICATION_JSON_UTF8).content(regularDriverJSON))
+					.andExpect(jsonPath("$.driver.licensePlate", is(regularDriver.getLicensePlate())))
+					.andExpect(jsonPath("$.end").isEmpty()).andExpect(status().isOk());
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	@Test
+	public void shouldStartParkingMeterDisabledDriver() {
+
+		try {
+			mvc.perform(post("/api/start").contentType(MediaType.APPLICATION_JSON_UTF8).content(disabledDriverJSON))
+					.andExpect(jsonPath("$.driver.licensePlate", is(disabledDriver.getLicensePlate())))
+					.andExpect(jsonPath("$.end").isEmpty()).andExpect(status().isOk());
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
dlaczego tak w każdym teście? Takiego łapania wyjątków nie powinno tu być

Reply:
Zdecydowanie powinno być _throws_. Oczywiście, jeśli jest bardziej eleganckie rozwiązanie, to proszę o poprawienie mnie.